examples/Planning/GenerateKeywordIdeas.php

```diff
@@ -0,0 +1,211 @@
+
+        // Make sure that keywords and/or page URL were specified. The request must have exactly one
+        // of urlSeed, keywordSeed, or keywordAndUrlSeed set.
+        if (empty($keywords) && is_null($pageUrl)) {
+            throw new \InvalidArgumentException(
+                'At least one of keywords or page URL is required, but neither was specified.'
+            );
+        }
+
+        // Specify the optional arguments of the request as a keywordSeed, UrlSeed,
+        // or KeywordAndUrlSeed.
+        $requestOptionalArgs = [];
+        if (empty($keywords)) {
+            // Only page URL was specified, so use a UrlSeed.
+            $requestOptionalArgs['urlSeed'] =
+                new UrlSeed(['url' => new StringValue(['value' => $pageUrl])]);
+        } elseif (is_null($pageUrl)) {
+            // Only keywords were specified, so use a KeywordSeed.
+            $requestOptionalArgs['keywordSeed'] = new KeywordSeed([
+                'keywords' => array_map(function ($keyword) {
+                    return new StringValue(['value' => $keyword]);
+                }, $keywords)
+            ]);
+        } else {
+            // Both page URL and keywords were specified, so use a KeywordAndUrlSeed.
+            $requestOptionalArgs['keywordAndUrlSeed'] = new KeywordAndUrlSeed([
+                'url' => new StringValue(['value' => $pageUrl]),
+                'keywords' => array_map(function ($keyword) {
+                    return new StringValue(['value' => $keyword]);
+                }, $keywords)
+            ]);
+        }
+
+        // Create a list of geo target constants based on the resource name of specified location
+        // IDs.
+        $geoTargetConstants =  array_map(function ($locationId) {
+            return new StringValue(
+                ['value' => ResourceNames::forGeoTargetConstant($locationId)]
+            );
+        }, $locationIds);
+
+        // Generate keyword ideas based on the specified parameters.
+        $response = $keywordPlanIdeaServiceClient->generateKeywordIdeas(
+            $customerId,
+            // Set the language resource using the provided language ID.
+            new StringValue(['value' => ResourceNames::forLanguageConstant($languageId)]),
+            // Add the resource name of each location ID to the request.
+            $geoTargetConstants,
+            // Set the network. To restrict to only Google Search, change the parameter below to
+            // KeywordPlanNetwork::GOOGLE_SEARCH.
+            KeywordPlanNetwork::GOOGLE_SEARCH_AND_PARTNERS,
+            $requestOptionalArgs
+        );
+
+        // Iterate over the results and print its detail.
+        foreach ($response->getResults() as $result) {
+            /** @var GenerateKeywordIdeaResult $result */
+            // Note that the competition printed below is enum value.
+            // For example, a value of 2 will be returned when the competition is 'LOW'.
```

Comment:
Thanet, could you pls file a bug against the code generator component to make the rendering of this value better?

Reply:
This has been in my radar. 
The proto lib will release a util that helps this soon already. It should be launched soon next year.